sshd/auth.go

```diff
@@ -5,17 +5,26 @@ import (
 	"encoding/base64"
 	"errors"
 	"net"
+	"time"
 
 	"github.com/shazow/ssh-chat/internal/sanitize"
 	"golang.org/x/crypto/ssh"
 )
 
-// Auth is used to authenticate connections based on public keys.
+// Auth is used to authenticate connections.
 type Auth interface {
 	// Whether to allow connections without a public key.
 	AllowAnonymous() bool
-	// Given address and public key and client agent string, returns nil if the connection should be allowed.
-	Check(net.Addr, ssh.PublicKey, string) error
+	// If passphrase authentication is accepted
+	AcceptPassphrase() bool
+	// Given address and public key and client agent string, returns nil if the connection is not banned.
+	CheckBans(net.Addr, ssh.PublicKey, string) error
+	// Given a public key, returns nil if the connection should be allowed.
+	CheckPubkey(ssh.PublicKey) error
```

Comment:
Sorry, one more nit rename: Let's call this CheckPublicKey for consistency.

Reply:
OK